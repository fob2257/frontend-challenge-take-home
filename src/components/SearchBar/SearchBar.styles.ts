import styled, { css } from 'styled-components';

const fontStyles = css`
  font-family: 'Manrope', sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  color: #3a434d;
  margin: 0;
`;

export const Form = styled.form`
  display: flex;
  justify-content: end;
  align-items: center;
  margin-bottom: 48px;
`;

export const Input = styled.input`
  ${fontStyles}
  padding: 8px 12px;
  width: 100%;
  height: 48px;
  background: #ffffff;
  border: 1px solid #c2c2c2;
`;

export const SubmitButton = styled.button.attrs(() => ({ type: 'submit' }))`
  background-color: unset;
  border: unset;
  cursor: pointer;
  position: absolute;
  padding-right: 16px;
`;
