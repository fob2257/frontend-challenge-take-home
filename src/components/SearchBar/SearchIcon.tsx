import React from 'react';

const SearchIcon = (props) => (
  <svg width={25} height={25} fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
    <circle
      cx={11.311}
      cy={11.31}
      r={8.056}
      transform="rotate(-23.025 11.31 11.31)"
      stroke="#3A434D"
      strokeWidth={1.5}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="m17.007 17.006 6.248 6.249"
      stroke="#3A434D"
      strokeWidth={1.5}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);

export default SearchIcon;
