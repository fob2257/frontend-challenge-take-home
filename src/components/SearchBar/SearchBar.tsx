import React from 'react';

import { Form, Input, SubmitButton } from './SearchBar.styles';
import SearchIcon from './SearchIcon';

interface Props {
  searchValue?: string;
  handleOnChange?: (searchTerm: string) => void;
}

const SearchBar: React.FC<Props> = ({
  searchValue = '',
  handleOnChange = () => {},
}) => {
  const handleFormOnSubmit = (e) => e.preventDefault();
  const handleInputOnChange = ({ target: { value } }) => handleOnChange(value);

  return (
    <Form aria-label="search form" onSubmit={handleFormOnSubmit}>
      <Input
        aria-label="search input"
        placeholder="Search by name"
        value={searchValue}
        onChange={handleInputOnChange}
      />
      <SubmitButton>
        <SearchIcon />
      </SubmitButton>
    </Form>
  );
};

export default SearchBar;
