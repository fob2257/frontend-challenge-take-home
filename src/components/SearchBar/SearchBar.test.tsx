import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';

import SearchBar from './SearchBar';

describe('<SearchBar />', () => {
  it('should render correctly', () => {
    render(<SearchBar />);

    const formElement = screen.getByRole('form', { name: 'search form' });

    expect(formElement).toBeInTheDocument();
  });

  it('should return typed value', () => {
    let searchValue = '';
    const mockedHandleOnChange = jest.fn((value) => {
      searchValue = value;
    });

    render(
      <SearchBar
        searchValue={searchValue}
        handleOnChange={mockedHandleOnChange}
      />,
    );

    const inputElement = screen.getByRole('textbox', { name: 'search input' });
    const value = 'python';
    fireEvent.change(inputElement, { target: { value } });

    expect(mockedHandleOnChange).toHaveBeenCalled();
    expect(searchValue).toBe(value);
  });

  it('should match snapshot', () => {
    const component = create(<SearchBar />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
