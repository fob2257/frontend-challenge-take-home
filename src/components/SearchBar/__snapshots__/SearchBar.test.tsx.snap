// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapshot 1`] = `
<form
  aria-label="search form"
  className="sc-bczRLJ fmGyHp"
  onSubmit={[Function]}
>
  <input
    aria-label="search input"
    className="sc-gsnTZi gcnwvc"
    onChange={[Function]}
    placeholder="Search by name"
    value=""
  />
  <button
    className="sc-dkzDqf fdORFM"
    type="submit"
  >
    <svg
      fill="none"
      height={25}
      width={25}
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx={11.311}
        cy={11.31}
        r={8.056}
        stroke="#3A434D"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        transform="rotate(-23.025 11.31 11.31)"
      />
      <path
        d="m17.007 17.006 6.248 6.249"
        stroke="#3A434D"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
      />
    </svg>
  </button>
</form>
`;
